lib/cache/loadable.go

```diff
@@ -60,9 +66,21 @@ func (c *LoadableCache[T]) Get(ctx context.Context, key any) (T, error) {
 	}
 
 	// Unable to find in cache, try to load it from load function
-	object, err = c.loadFunc(ctx, key)
-	if err != nil {
-		return object, err
+	var r any
+	if r, err, _ = c.singleFlight.Do(
+		c.getCacheKey(key),
+		func() (any, error) {
+			return c.loadFunc(ctx, key)
+		},
+	); err != nil {
+		return *new(T), err
+	}
+	var ok bool
+	if object, ok = r.(T); !ok {
+		zero := *new(T)
+		return zero, errors.New(
+			fmt.Sprintf("returned value can't be cast to %T", zero),
+		)
 	}
```

Comment:
```suggestion
	cacheKey := c.getCacheKey(key)
	zero := *new(T)

	loadedResult, err, _ := c.singleFlight.Do(
		cacheKey,
		func() (any, error) {
			return c.loadFunc(ctx, key)
		},
	)
	if err != nil {
		return zero, err
	}

	var ok bool
	if object, ok = loadedResult.(T); !ok {
		return zero, errors.New(
			fmt.Sprintf("returned value can't be cast to %T", zero),
		)
	}
```

Reply:
I don't like the `r` variable name, elsewhere, I am ok with the code, thank youè